### ISSUE TITLE
Compile-time error: "Argument field or value named 'order_by' can not be coerced"

### ISSUE THREAD

**Issue by Yakimych** [fsprojects/FSharp.Data.GraphQL#270]
### Description

I am using Hasura, and trying to run the following query:
```
let operation = MyProvider.Operation<"""query q {
      test_table (order_by: {some_name: asc}) {
        id
        some_name
      }
    }""">
```

The query is valid and works if run manually, but I get a compile-time error with the F# Type Provider:
```
error FS3033: The type provider 'FSharp.Data.GraphQL.Client.GraphQLTypeProvider'
reported an error:
Argument field or value named 'order_by' can not be coerced.
It does not match a valid literal representation for the type.
Path: ["q"; "test_table"]
```

The relevant part of the schema:
```
{
  "inputFields": null,
  "kind": "ENUM",
  "possibleTypes": null,
  "interfaces": null,
  "name": "order_by",
  "enumValues": [
    {
      "isDeprecated": false,
      "deprecationReason": null,
      "name": "asc",
      "description": "in the ascending order, nulls last"
    },
  ...
  ],
  "description": "column ordering options",
  "fields": null
}
```

I've tried other parameters, such as `offset`, `limit`, `where` (this one works now: https://github.com/fsprojects/FSharp.Data.GraphQL/issues/263), and they work. `distinct_on`, however, results in the same compile time error. Could this have something to do with the `_` in the name?

### Repro steps

See description.

### Expected behavior

The application compiles, the query returns the same result as when run manually.

### Actual behavior

Compile-time error.

### Known workarounds

None.

### Related information 

* Mac OS X
* Version 1.0.3
* .NET Core 3.1

**Comment by johnberzy-bazinga** [comment]
By any chance do you have any other query named `q`?

**Comment by Yakimych** [comment]
Nope, this one is the only one.
I am pretty much testing "in place". The query from https://github.com/fsprojects/FSharp.Data.GraphQL/issues/263 compiles and works now - if I replace the contents of parentheses with e.g. `(offset: 1, limit: 2)`, everything works. If I replace it with `order_by: ...` or `distinct_on: ...`, it fails to compile immediately with the "...cannot be coerced...` error.

**Comment by johnberzy-bazinga** [comment]
@Yakimych Do you mind sending me the schema dump? If it happens at compile-time I can try to debug without needing to run a server.

**Comment by Yakimych** [comment]
Yes, I wasn't sure I wanted to dump 3000 lines of code, but apparently I can upload a json-file if I rename it to txt 🙂 
[schema.txt](https://github.com/fsprojects/FSharp.Data.GraphQL/files/4371534/schema.txt)

**Comment by Yakimych** [comment]
@johnberzy-bazinga Got the same problem when trying to create a mutation against the same schema:
```
[FS3033] The type provider 'FSharp.Data.GraphQL.Client.GraphQLTypeProvider'
reported an error:
Argument field or value named 'objects' can not be coerced.
It does not match a valid literal representation for the type.
Path: ["addPlayer"; "insert_players"]
```

The mutation itself:
```
  mutation addPlayer {
    insert_players(objects: {name: "somename", communityId: 1}) {
      returning {
        id
      }
    }
  }
```

**Comment by treet** [comment]
I had the exact same problem, a workaround is to wrap the order_by value in a list (credits to @Yakimych):

Does not work: `order_by: {some_name: asc}`
Works: `order_by: [{some_name: asc}]`

This appears to be a bug in how FSharp.Data.GraphQL does input coercion: 

![image](https://user-images.githubusercontent.com/1241773/84577827-ede83b80-adbf-11ea-9e9e-9955862e4afb.png)
